### PR TITLE
Add auto-generated product layout

### LINKED ISSUE #120: Add auto-rendered product layout
Add auto-rendered product layout where the product data is rendered based on [product drop](https://www.voog.com/developers/markup/ecommerce-objects/product) data.

It should look like current "Product" layout. Use new [image](https://www.voog.com/developers/markup/tags/image) or [image_data](https://www.voog.com/developers/markup/tags/image-data) tag for product cover image.

It content type must be `product`:

```json
{
  "content_type": "product",
  "component": false,
  "file": "layouts/auto_generated_product.tpl",
  "layout_name": "auto_generated_product",
  "title": "Auto-generated product"
}
```

See also buy button rendering example in [Nuuk design](https://github.com/Voog/design-nuuk/blob/master/components/buy-button.tpl).

## TODO

* [ ] Add "Auto-generated product" layout
* [ ] Check and improve "[product list](https://www.voog.com/support/online-store/product-list-content-area)" widget styles when needed.

### PR DESCRIPTION
**Pull request by pxska**
* Add auto-generated product layout
* Add browser hack for Chrome to render crispier images

Closes #120.

### CODE REVIEW

File: layouts/auto_generated_product.tpl [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+          <div class="flex_row-2--item-50 js-product-content-right">
+            <div class="mar_0-32 flex_col t-sticky">
+              <div class="content-formatted content-item-title" data-search-indexing-allowed="true">
+                <h1>{%- editable product.name -%}</h1>
+              </div>
+
+              <div class="product-price">
+                {%- if product.price_max_with_tax != product.price_min_with_tax -%}
+                  {{ product.price_min_with_tax | money_with_currency: product.currency -}}
+                  <span class="product-price-divider">–</span>
```
Comment (PeeterPaal) [inline layouts/auto_generated_product.tpl:72]: Perhaps let's add some left and right margin for `product-price-divider`

**Review by PeeterPaal — CHANGES_REQUESTED** [review]
Nice! Some comments:

- [ ] Maybe consider adding some margin for the title, price and description.
Screenshot 2022-02-09 at 14 09 16

File: layouts/auto_generated_product.tpl [hunk 2]
```diff
@@ -0,0 +1,119 @@
+<!DOCTYPE html>
+{% include "template-variables" %}
+
+<html class="{% if editmode %}editmode{% else %}public{% endif %}" lang="{{ page.language_code }}" data-view-state="{{ view_mode }}">
```
Comment (tanelj) [inline layouts/auto_generated_product.tpl:4]: There was all ready class `editmode` for `html` tag and there is no need to duplicate it to `data-view-state`.

File: sources/javascripts/concat/global/main.js [hunk 3]
```diff
@@ -255,6 +255,39 @@
     }
   };
 
+
```
Comment (tanelj) [inline sources/javascripts/concat/global/main.js:258]: NB! Now there has two `handleProductPageContent` functions! Please unify!

File: sources/javascripts/concat/global/main.js [hunk 4]
```diff
@@ -265,6 +298,10 @@
     handleProductPageContent:handleProductPageContent
   });
 
+  window.template = $.extend(window.template || {}, {
+    handleProductPageContent: handleProductPageContent
+  })
+
```
Comment (tanelj) [inline sources/javascripts/concat/global/main.js:304]: `handleProductPageContent` is all ready part of `window.site`.

File: layouts/auto_generated_product.tpl [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+    {% include "footer" %}
+  </div>
+
+  {% include "site-signout" %}
+  {% include "javascripts" %}
+  {% include "edicy-tools" %}
+
+  <script>
+    template && template.handleProductPageContent();
```
Comment (tanelj) [inline layouts/auto_generated_product.tpl:113]: `handleProductPageContent` is all ready `window.site’ scope.

File: sources/stylesheets/components/_image-drop-area.scss [hunk 6]
```diff
@@ -7,6 +7,10 @@
   overflow: hidden;
   margin-top: 10px;
 
+  [data-view-state="editmode"] & {
```
Comment (tanelj) [inline sources/stylesheets/components/_image-drop-area.scss:10]: We have all ready `html.editmode` class.

File: components/template-variables.tpl [hunk 7]
```diff
@@ -0,0 +1,8 @@
+{% capture dont_render %}
```
Comment (tanelj) [inline components/template-variables.tpl:1]: We have allready `html.editmode` class.

**Review by tanelj — CHANGES_REQUESTED** [review]
There has some double code.

Also:
* [ ] Out of stock buy button wraps to two lines in Safari (v15.2) preview mode.
Screenshot 2022-02-10 at 10 59 23

**Review by tanelj — APPROVED** [review]
Nice work!

**Review by PeeterPaal — CHANGES_REQUESTED** [review]
- [ ] Product gallery is currently not moved under the buy button in mobile view.

**Review by PeeterPaal — APPROVED** [review]
Nice!

**Pull request merged**